### PR TITLE
Yegan: Task 497-504: esnli Dataset and refresd Dataset

### PR DESCRIPTION
**Pull request by ilobo22**

### CODE REVIEW

**Comment by swarooprm** [comment]
@yeganehkordi, wondering if you can help review these tasks.

File: tasks/README.md [hunk 1]
```diff
@@ -6,4 +6,9 @@ This directory contains the tasks that are part of this benchmark.
 Name | Summary | Category
 ---- | ----------- | --------
 `task1001_squad1.1_question_generation` | Generating guestions (based on SQuAD 1.1) | Question Generation  
-`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
\ No newline at end of file
+`task1002_squad1.1_answer_generation` | Generating answers to SQuAD 1.1 questions | Answer Generation
+`task497_esnli_classification` | Classification based on a premise and hypothesis | Classification
```
Comment (yeganehkordi) [inline tasks/README.md:10]: Please elaborate more on task 497.

File: tasks/task497_esnli_classification.json [hunk 2]
```diff
@@ -0,0 +1,444 @@
+{
+  "Contributors": ["Ian Lobo"],
+  "Source": [
+    "e-SNLI Dataset https://github.com/OanaMariaCamburu/e-SNLI/blob/master/dataset/esnli_test.csv"
+  ],
+  "Categories": ["Classification"],
+  "Definition": "Given Sentence 1 (the premise) and then given Sentence 2 (the hypothesis), determine if the hypothesis can be concluded from the premise and assigning it a label among entailment, neutral, or contradiction.",
```
Comment (yeganehkordi) [inline tasks/task497_esnli_classification.json:7]: Please explain more about the entailment, neutral, and contradiction in the definition.

File: tasks/task499_esnli_classification.json [hunk 3]
```diff
@@ -0,0 +1,1234 @@
+{
+  "Contributors": ["Ian Lobo"],
+  "Source": [
+    "e-SNLI Dataset https://github.com/OanaMariaCamburu/e-SNLI/blob/master/dataset/esnli_test.csv, https://cims.nyu.edu/~sbowman/multinli/paper.pdf"
+  ],
+  "Categories": ["Classification"],
+  "Definition": "Given Sentence 1 and Sentence 2, determine if the two sentences clearly agree/disagree with each other, or if it can't be determined. Indicate your answer as yes or no respectively.",
```
Comment (yeganehkordi) [inline tasks/task499_esnli_classification.json:7]: Maybe add yes means the two sentences clearly agree/disagree with each other.

**Review by yeganehkordi — COMMENTED** [review]
Thanks for your work! Some comments:
- I don't understand, what's the difference between tasks 497 and 498?
- I think it would be better if you separate the sentences with <sep>.

**Comment by ilobo22** [comment]
Regarding your first comment, there is not much of a difference. I did it this way because the allenai tasks did it this way with a similar dataset. 
Regarding your second comment, could you show me an example with the "<sep>" input?

**Review by yeganehkordi — COMMENTED** [review]
Thanks!
In this case, I think it's better to split the dataset and use separate instances for tasks 497 and 498.
Yes, for example, you can use this format for inputs of task 497: `This church choir sings to the masses as they sing joyous songs from the book at a church. <sep> The church has cracks in the ceiling.`

**Comment by ilobo22** [comment]
Hi Yeganeh! So I added the <sep> value as you asked. Couple of questions before we close this PR: 1) What did you mean by "split the dataset and use separate instances for tasks 497 and 498"? Can I just go ahead and delete task 498 since its similar to 497 and 2) Should I include the <sep> value for tasks 500 and 501 even though it deals with french and english sentences?

**Review by yeganehkordi — COMMENTED** [review]
Hi, Thanks for your great job!
1) You can split the dataset into train and dev and use the train split for task 497 and the dev for task 498, or vice versa. 
Yes, you can delete one of them. But I prefer to keep them both because there is a slight difference in their definition. (the first task uses premise and hypothesis)
2) Yes, it's useful for task 500, but we don't need `<sep>` in task 501. As a rule of thumb, you need to use a separator token whenever you have more than one input(e.g., two sentences). This separator can be `<sep>` or `\n`. Here we prefer to use `<sep> because it's a non-scape token.

**Comment by swarooprm** [comment]
@ilobo22, can you address reviewer comments and let @yeganehkordi know once you do that so that she can review one final time and let us know if she is happy or has any other feedback?

**Comment by ilobo22** [comment]
Hi Yeganeh. I did everything you asked however, for the train and dev split, I will have to rework my code because I got my dataset from .csv in github and didn't use huggingface's library.

**Comment by yeganehkordi** [comment]
Hi @ilobo22, Everything looks good to me! Just one minor comment: Please add the appropriate fields to the multilingual tasks based on: https://github.com/allenai/natural-instructions-expansion/issues/113
@swarooprm Task 497 and 498 are too similar, but I think we can keep both of them. Could you please take a look at them? Overall, I'm happy with these tasks.

**Review by yeganehkordi — APPROVED** [review]

**Comment by swarooprm** [comment]
Thanks @yeganehkordi for the review. Task 497 and 498 are not unique, but we count best 4 tasks, so this is fine.

**Comment by swarooprm** [comment]
Your HW2 is complete now.

Would you like to submit the same tasks to our big project repo https://github.com/allenai/natural-instructions-expansion/pulls? You just need to fork that repo, rename tasks starting from the latest number in the repo and create PR there. This way, you can be part of our research on "learning from instructions instead of learning from examples (conventional machine learning)". This is purely optional and your HW2 grade does not depend on this @ilobo22

**Comment by ilobo22** [comment]
Done!

**Pull request closed without merging**